tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_selectors.ts

```diff
@@ -307,3 +308,10 @@ export const getSourceFileList = createSelector(
     return state.sourceCode.sourceFileList;
   }
 );
+
+export const getSourceFileContents = createSelector(
```

Comment:
Without this selector, views are unaware that the sourceFileContents relies on having indices aligned with sourceFileList.  I think that is a nice property, because requiring 2 arrays be kept in sync is unfortunate.  Exposing this makes it easy for views and others to reuse an array index from sourceFileList, and access the corresponding file contents.

Can we avoid exporting the list of file contents, and instead just expose a selector that takes a specific {host name, file name, (run?)} argument?

Reply:
Good point. I thought about this and decided to address this comment as follows:

1. Add a new action (`sourceLineFocused`) for the action of focusing on a file in the list of source files. In follow-up CLs, this action will be fired when the user clicks a stack frame (e.g., a line of a source file).
2. Modify the new effect to trigger on the `sourceLineFocused` action, instead of the `sourceFileRequested` action.
3. Add a reducer for the said action, which populates the `focusLineSpec` field in the store state.
4. Remove this `getSourceFileContents` selector. Instead add two selectors `getFocusedSourceFileIndex` and `getFocusedSourceFileContent`, which are used by the aforementioned state. This way, the selectors will not expose the entire list of file contents.